Standards/scs-XXXX-vN-key-manager-standard.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Key-Manager Standard
+type: Standard
+status: Draft
+track: IaaS
+---
+
+## Introduction
+
+To encrypt user data like volumes or in the future also Images and ephemeral storage for VMs, the key is needed to be present in the infrastructure.
+Therefore a key-manager is needed to store the keys and apply authorization policies on requests.
+The OpenStack key-manager implemenation that can bu utilized for this is named Barbican.
+This standard aims to provide a base level of security for Cloud Service Providers that integrate a key-manager into their deployments.
+
+## Terminology
+
+| Term | Meaning |
+|---|---|
+| API | Application Programming Interface, often referring to the REST API interfaces provided by OpenStack and related services |
+| CSP | Cloud Service Provider, provider managing the OpenStack infrastructure |
+| IaaS | Infrastructure-as-a-Service |
+| HSM | Hardware Security Module |
+| KEK | Key Encryption Key |
+
+## Motivation
+
+When user data is encrypted the keys need to be stored securely in the Infrastructure to be present, when a possible encryption or decryption needs to take place.
+Key-managers like Barbican provide this functionality on the IaaS-Level.
+Using such a key-manger and thus provide to customers the possibility to encrypt user data is not yet implemented everywhere in IaaS deployments.
+A first step towards more security is to encourage CSPs to use this service and provide the feature of using encrypted resources to customers.
+It is also important to take a closer look into the key-manager and to apply aim for an appropiate level of security there.
+The Key-Manager is responsible for storing the keys securely, which can be done by encrypting them with a KEK and storing them in a database and store the KEK either also encrypted in the database or to store it somewhere else.
+Barbican as the OpenStack implementation of a key-manager is relying on a Master-KEK, which encrypts project-specific KEKs, which encrypt the Keys within a project.
```

Comment:
Barbican stores keys encrypted with the project specific KEK, including the KEK itself, in the database. Master KEK, used to encrypt project specific KEK is not stored in the database and is stored differently depending on the backend storage plugin used.